dstep/translator/MacroIndex.d

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright: Copyright (c) 2016 Wojciech Szęszoł. All rights reserved.
+ * Authors: Wojciech Szęszoł
+ * Version: Initial created: Mar 08, 2016
+ * License: $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost Software License 1.0)
+ */
+
+module dstep.translator.MacroIndex;
+
+import std.container.rbtree;
+
+import clang.c.Index;
+import clang.Cursor;
+import clang.SourceLocation;
+import clang.SourceRange;
+import clang.TranslationUnit;
+
+class InvalidArgumentError : object.Error
+{
+    this (string message, string file = __FILE__, ulong line = __LINE__)
+    {
+        super(message, file, line);
+    }
+}
+
+class MacroIndex
+{
+    private static bool pred(in Cursor a, in Cursor b)
+    {
+        return SourceLocation.lexicalLess(a.location, b.location);
+    }
+
+    private TranslationUnit unit;
+    private alias CursorRedBlackTree = RedBlackTree!(Cursor, (in Cursor a, in Cursor b) => pred(a, b));
```

Comment:
I wanted to do that but then the compiler complains with this:

    dstep/translator/MacroIndex.d(34,40): Error: template instance std.container.rbtree.RedBlackTree!(Cursor, pred) does not match template declaration RedBlackTree(T, alias less = "a < b", bool allowDuplicates = false) if (is(typeof(binaryFun!less(T.init, T.init))))

And I do not know why really : /